packages/bitcore-client/src/storage/mongo.ts

```diff
@@ -0,0 +1,146 @@
+import 'source-map-support/register'
+import { MongoClient } from 'mongodb'
+import {baToJSON} from "ethereumjs-util";
+import {Transform} from "stream";
+
+export class Mongo {
+  path: string;
+  db: any;
+  collectionName: string;
+  collection: any;
+  errorIfExists?: boolean;
+  createIfMissing: boolean;
+  dbName: string;
+  databaseName: string;
+  client: any;
+  port: string;
+  addressCollectionName: string;
+  constructor(params: {
+    path?: string;
+    createIfMissing: boolean;
+    errorIfExists: boolean;
+  }) {
+    const { path, createIfMissing, errorIfExists } = params;
+    if (!path) {
+      throw new Error('Must specify a mongo url as path');
+    }
+    this.port = '27017';
+    this.path = path + ':' + this.port;
+    this.createIfMissing = createIfMissing;
+    this.errorIfExists = errorIfExists;
+    let databasePath = path.split('/');
+    if (databasePath[databasePath.length -1] === '') {
+      databasePath.pop();
+    }
+    this.databaseName = databasePath.pop();
+    this.collectionName = 'bitcoreWallets';
+    this.addressCollectionName = 'bitcoreWalletAddresses';
+  }
+
+  async init(params) {
+    const { wallet, addresses } = params;
+    try {
+      this.client = new MongoClient(this.path, { useNewUrlParser: true });
+      await this.client.connect();
+      console.log('connected');
+      this.db = this.client.db(this.databaseName);
+      if (wallet) {
+        this.collection = this.db.collection(this.collectionName);
+      } else if (addresses) {
+        this.collection = this.db.collection(this.addressCollectionName)
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  async close() {
+    await this.client.close();
+    console.log('closed connection');
+  }
+
+  async listWallets() {
+    await this.init({wallet: 1});
+    const stream = new Transform({
+      objectMode: true,
+      transform(data, enc, next) {
+        this.push(JSON.stringify(data));
+        next();
+      }
+    });
+    const cursor = this.collection.find({'name': { '$exists': true }}, { 'name': 1, 'chain': 1, 'network': 1 }).pipe(stream);
```

Comment:
likely want to index on name

something like createIndex({name: 1}) should probably be in init